### PR TITLE
Switch to golang native error wrapping

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Replacing the github.com/pkg/errors dependency with native wrapping.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@haircommander PTAL, this PR has high potential for merge conflicts

File: internal/config/nsmgr/nsmgr.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func (mgr *NamespaceManager) NewPodNamespaces(cfg *PodNamespacesConfig) ([]Names
 	for _, ns := range cfg.Namespaces {
 		arg, ok := typeToArg[ns.Type]
 		if !ok {
-			return nil, errors.Errorf("Invalid namespace type: %s", ns.Type)
+			return nil, fmt.Errorf("Invalid namespace type: %s", ns.Type)
```
[the remark this answers is not in the document]
Comment (haircommander) [reply internal/config/nsmgr/nsmgr.go:101]: would you mind updating these @saschagrunert

[the remark this answers is not in the document]
Comment (saschagrunert) [reply internal/config/nsmgr/nsmgr.go]: Done :+1:

**Comment by saschagrunert** [comment]
/retest-required

@haircommander this would need another LGTM :)

**Comment by haircommander** [comment]
/lgtm

thanks!

**Pull request merged**